File: Rains of Reason/variableName.js
/**
 * 
 * @param {string} name 
 */
function solution(name) {
 return /^[a-z_][\w_]*$/i.test(name)
return /^[a-zA-Z_][a-zA-Z0-9_]*$/.test(name)
}

console.log(solution('1dol'));

File: Rains of Reason/variableName.py
import re

def solution(name:str):
    return bool(re.match(r'^[a-z_]\w*$', name, re.IGNORECASE)) or bool(re.match(r'^[a-zA-Z_][a-zA-Z0-9_]*$', name))

print(solution('1dol'))